python/llama-index-server/llama_index/server/tools/index/citation.py

```
@@ -0,0 +1,81 @@
+class CitationSynthesizer(Accumulate):
+    """
+    Overload the Accumulate synthesizer to:
+    1. Update prepare node metadata for citation id
+    2. Update text_qa_template to include citations
+    """
+
+    def __init__(self, *args, **kwargs):
+        text_qa_template = kwargs.pop("text_qa_template", None)
+        if text_qa_template is None:
+            text_qa_template = PromptTemplate(template=CITATION_PROMPT)
+        super().__init__(text_qa_template=text_qa_template, *args, **kwargs)
```

Ruff (0.11.9)

Star-arg unpacking after a keyword argument is strongly discouraged

(B026)